doc: nrf: Added page about Matter ecosystems certification (#12546)
nrfconnect/sdk-nrf: doc/nrf/protocols/matter/end_product/ecosystems_certification.rst

.. _ug_matter_ecosystems_certification:

Ecosystems certification
########################

.. contents::
   :local:
   :depth: 2

The Matter stack provided in the |NCS| works with commercial ecosystems that are compatible with official Matter implementation.
The Matter product is called compatible with Matter implementation once it passed relevant Matter certification.

> However some of the ecosystem providers require passing additional certification to verify that Matter product is able to work with their applications and support their features.

Reviewer: I think the word "require" here and in line 21 is a bit too strict (e.g. Google cert is not mandatory afaik). I think we can say that ecosystems offer additional certification programs to ensure that the product works well with the given ecosystem or correctly implement additional proprietary to matter features.

Author: Fair point, applied.

Once the Matter product passes the ecosystem related certification, the appropriate information can be placed on the product's packaging.

.. _ug_matter_google_certification:

Works with Google Home certification
************************************

The Google Home runs the `Works with Google Home`_ certification program to ensure the Matter product supports Matter features in the same extent, as the Google Home application.
The product that passed all the certification test cases is allowed to use the Works with Google Home badge.

The Matter samples delivered in the |NCS| did not receive official certificates, as these can be obtained only for the final products, however they were tested against the Works with Google Home certification test cases and they successfully passed all of them.
The following list contains a full list of Matter samples that were verified to pass Works with Google certification test cases:

* :ref:`Matter door lock <matter_lock_sample>`
* :ref:`Matter light bulb <matter_light_bulb_sample>`
* :ref:`Matter thermostat <matter_thermostat_sample>`
* :ref:`Matter window covering <matter_window_covering_sample>`

> * :ref:`Matter weather station <matter_weather_station_app>`

Author: No, I didn't think about that. Fixed.

.. _ug_matter_amazon_certification:

Amazon Frustration-Free Setup certification
*******************************************

Frustration-Free Setup (FFS) is Amazon's proprietary feature that uses Amazon cloud service to provide onboarding information necessary to commission the Matter device to a Matter fabric.

> The commissioning process is automatically triggered by the Amazon commissioner after detecting accessory's Bluetooth LE advertising messages supporting proprietary format and it does not require any user interaction.

Reviewer 2: ```suggestion
The commissioning process is automatically triggered by the Amazon commissioner and it does not require any user interaction. 
The process starts when the commissioner detects the accessory's Bluetooth LE advertising messages in the proprietary format.
```

For more information about the FFS and how to enable it in Matter samples, see :ref:`Amazon Frustration-Free Setup configuration<ug_matter_configuring_ffs>` page.

FFS requires passing the additional certification test cases for Matter called `Matter Simple Setup`_.

> There are following two types of certification:

Reviewer 2: ```suggestion
There are two types of certification:
```

> * Default - for standard products or for the creation of a reference design
> * Simplified - for products using a certified reference design

Reviewer 2: ```suggestion
* Default - For standard products or for the creation of a reference design.
* Simplified - For products using a certified reference design.
```

The Matter platform provided in the |NCS| was tested against the FFS certification test cases and passed them in several variants.

> If you create a Matter product based on |NCS|, you can benefit from it and approach the simplified certification path.

Reviewer 2: ```suggestion
This means that if you create a Matter product based on the |NCS|, you can approach the simplified certification path.
```

> If you create a Matter product based on |NCS|, you can benefit from it and approach the simplified certification path.

Reviewer: Could we please link to the Amazon certification page? Since it would be nice to understand what "simplified" means - i.e. less testing, no need to send devices to Amazon for lab testing.

Author: We link to the Amazon certification page in this sentence `FFS requires passing the additional certification test cases for Matter called `Matter Simple Setup`_.` Do you think it is enough or should we point to some specific part of the page?

> You can do that by filling the Reference APID field, during the product registration on the Amazon's developer page with one of the following values:

Reviewer 2: ```suggestion
You can do this by filling the Reference APID field when you register your product on Amazon's developer page.
Use one of the following values:
```

+-----------------------------------------------+-----------------------+

> | Matter platform variant                       | Amazon Reference APID |

Reviewer: What about... Amazon Reference APID -> Reference Design APID ?

FFS requires passing the additional certification test cases for Matter called `Matter Simple Setup`_.
There are two types of certification:

* Default - For standard products or for the creation of a reference design.
* Simplified - For products using a certified reference design.

The Matter platform provided in the |NCS| was tested against the FFS certification test cases and passed them in several variants.
This means that if you create a Matter product based on the |NCS|, you can approach the simplified certification path.
You can do this by filling the Reference APID field when you register your product on Amazon's developer page.
Use one of the following values:

+-----------------------------------------------+-----------------------+
| Matter platform variant                       | Reference Design APID |
+===============================================+=======================+
| nrf52840dk_nrf52840 (Matter over Thread)      | ZNwt                  |
+-----------------------------------------------+-----------------------+
| nrf5340dk_nrf5340_cpuapp (Matter over Thread) | xzNd                  |
+-----------------------------------------------+-----------------------+

> | nrf7002dk_nrf5340_cpuapp (Matter over Wi-Fi)  | jyjh                  |

Reviewer 2: Should these be board names (in which case the `_cpuapp` part should not be included) or build targets (in which case they should be formatted `nrf52840dk_nrf52840`). For refrence from the the Light bulb sample:
![image](https://github.com/nrfconnect/sdk-nrf/assets/83582261/056c98ab-32ac-44c9-a8c0-1fa61ee5136e)

Author: Fair point, actually I think it should be hardware platforms. I updated the table with this change.
